mitiq/pec/pec.py

```diff
@@ -134,10 +134,13 @@ def execute_with_pec(
     collected_executor = generate_collected_executor(
         executor, force_run_all=force_run_all
     )
-    exp_values = collected_executor(sampled_circuits)
+    results = collected_executor(sampled_circuits)
 
     # Evaluate unbiased estimators [Temme2017] [Endo2018] [Takagi2020]
-    unbiased_estimators = [norm * s * val for s, val in zip(signs, exp_values)]
+    unbiased_estimators = [
+        norm * s * val  # type: ignore[operator]
+        for s, val in zip(signs, results)
+    ]
```

Comment:
I see that you renamed `exp_values` to `results`. This is good for me, but this doesn't mean that PEC now works with an executor that returns a `MeasurementResult`. Right? Just to double-check if I am following.

Reply:
Yes that's right. mypy complained about multiplying measurements by floats here (its awesome we have mypy now, I never would have caught this otherwise). PEC does not work with executors that return measurements, but nothing does for now and won't until observables are fully supported so that we can take measurements -> expectation values.